src/Donations/PaymentMethods/PaymentFunctions.ts

```diff
@@ -212,54 +223,74 @@ export function createDonationData({
-    };
-  }
+  let donationData: DonationRequestData;
 
-  if (isGift) {
-    if (giftDetails.type === "invitation") {
-      donationData = {
-        ...donationData,
-        ...{
-          gift: {
-            type: "invitation",
-            recipientName: giftDetails.recipientName,
-            recipientEmail: giftDetails.recipientEmail,
-            message: giftDetails.message,
-          },
+  // Handle supported donations with lineItems
+  if (isSupportedDonation && supportedProjectId && getDonationBreakdown) {
+    const { mainProjectQuantity, supportAmount, mainProjectAmount } =
+      getDonationBreakdown();
+
+    donationData = {
+      ...baseDonationFields,
+      purpose: "composite",
+      lineItems: [
+        {
+          project: projectDetails.id,
+          units: mainProjectQuantity,
+          amount: mainProjectAmount,
         },
-      };
-    } else if (giftDetails.type === "direct") {
-      donationData = {
-        ...donationData,
-        ...{
-          gift: {
-            type: "direct",
-            recipient: giftDetails.recipient,
-          },
+        {
+          project: supportedProjectId,
+          amount: supportAmount,
         },
-      };
+      ],
+    } satisfies CompositeDonationRequest;
+  } else {
+    // Add gift details if applicable
+    let _giftDetails: SentGift | undefined = undefined;
+    if (isGift) {
+      if (giftDetails.type === "invitation") {
+        _giftDetails = {
+          type: "invitation",
+          recipientName: giftDetails.recipientName,
+          recipientEmail: giftDetails.recipientEmail,
+          message: giftDetails.message,
+        };
+      } else if (giftDetails.type === "direct") {
+        _giftDetails = {
+          type: "direct",
+          recipient: giftDetails.recipient,
+        };
+      }
     }
+
+    // Handle regular donations
+    donationData = {
+      ...baseDonationFields,
+      purpose: projectDetails?.purpose,
+      project: projectDetails.id,
+      amount:
+        Number.isFinite(paymentSetup.unitCost) && Number.isFinite(quantity)
+          ? Math.round(paymentSetup.unitCost * quantity * 100) / 100
+          : (amount as number),
+      units: quantity,
+      ...(isGift && { gift: _giftDetails }),
+    } satisfies RegularDonationRequest;
   }
 
+  // Handle PlanetCash top-up special case
   if (projectDetails?.purpose === "planet-cash") {
-    // For PlanetCash Top-up
-
-    // No need to send quantity.
-    delete donationData.quantity;
-
-    // Since a user account can have only one planetCash account no need to send project (i.e planetCash account ID).
-    delete donationData.project;
-
-    //should not send gift details
-    delete donationData.gift;
+    // For PlanetCash Top-up, we need to modify the regular donation
+    if (donationData.purpose !== "composite") {
+      // Remove units and project for PlanetCash
+      delete (donationData as any).units;
+      delete (donationData as any).project;
```

Comment:
Can you use a proper type here instead of any?

Reply:
Need to refactor this in a separate PR